samples/ImplicitFlow/AureliaApp/src/open-id/open-id-response-parser.ts

```diff
@@ -0,0 +1,23 @@
+import { UriHelper } from "./uri-helper";
+import { OpenIdResponseModel } from "./open-id-response-model";
+
+export class OpenIdResponseParser {
+    public ParseWindowLocation(location: Location) : OpenIdResponseModel {
+        let hash = location.hash;
+
+        // turn the hash into a query string
+        // because that is what GetParameterByName expects.
+        let queryString = "&" + hash.substr(1);
+
+        let response: OpenIdResponseModel =
+            {
+                AccessToken: UriHelper.GetParameterByName("access_token", queryString),
+                ExpiresIn: UriHelper.GetParameterByName("expires_in", queryString),
+                IdToken: UriHelper.GetParameterByName("id_token", queryString),
+                State: UriHelper.GetParameterByName("state", queryString),
```

Comment:
In the `ClientModel`, I will make a note that validating the state is important for preventing XSRF attacks. There is probably a reason for the nonce too, eh?